common/js/src/popup-window/popup-opener.js

```diff
@@ -140,15 +149,32 @@ GSC.PopupOpener.runModalDialog = function(
     Object.assign(createWindowOptions, opt_createWindowOptionsOverrides);
   }
 
+  if (GSC.Packaging.MODE === GSC.Packaging.Mode.EXTENSION) {
+    lastUsedPopupId++;
+    opt_data['popup_id'] = lastUsedPopupId.toString();
+  }
+
   const promiseResolver = goog.Promise.withResolver();
 
+  // Set promiseResolver for SCC app mode
   const dataWithDialogCallbacks = {
-    'resolveModalDialog': promiseResolver.resolve,
-    'rejectModalDialog': promiseResolver.reject
+    'googleSmartCard_resolveModalDialog': promiseResolver.resolve,
+    'googleSmartCard_rejectModalDialog': promiseResolver.reject
   };
+
+  // Set promiseResolver for SCC extension mode
+  goog.global[`googleSmartCard_resolveModalDialog${lastUsedPopupId}`] =
+      promiseResolver.resolve;
+  goog.global[`googleSmartCard_rejectModalDialog${lastUsedPopupId}`] =
+      promiseResolver.reject;
+
   if (opt_data !== undefined)
     Object.assign(dataWithDialogCallbacks, opt_data);
 
+  if (GSC.Packaging.MODE === GSC.Packaging.Mode.EXTENSION && opt_data) {
+    url.searchParams.append('passed_data', JSON.stringify(opt_data));
```

Comment:
The bug with modifying the input `url` parameter is still here. Maybe it's still unclear for you what the issue is - let me illustrate it:

```
function f(url) {
  url.searchParams.append('passed_data', 'foo');
}
let someUrl = new URL('http://example.org');
f(someUrl);
console.log(someUrl.toString());
// prints: 'http://example.org/?passed_data=foo'
f(someUrl);
console.log(someUrl.toString());
// prints: 'http://example.org/?passed_data=foo&passed_data=foo'
```

As we can see from these results, the function is only modifying the "local" variable, but the changes end up in the caller's variable, which is totally wrong.

Regarding how to fix it - we need to make a copy of the `url` object. I guess this should work: `const modifiedUrl = new URL(url)`.

Reply:
ah ok, actually i understood the idea i just didn't see why we don't want to update the url itself (and not just the local variable), but the idea is that we create a new window with this modified url so that's why we don't need to change the actual current url